backend/app/src/main/kotlin/io/tolgee/security/JwtTokenProviderImpl.kt

```diff
@@ -26,31 +30,59 @@ class JwtTokenProviderImpl(
   private val configuration: TolgeeProperties,
   private val userAccountService: UserAccountService,
   private val jwtSecretProvider: JwtSecretProvider,
-  private val authenticationProvider: AuthenticationProvider
+  private val authenticationProvider: AuthenticationProvider,
+  private val currentDateProvider: CurrentDateProvider,
+  private val authenticationProperties: AuthenticationProperties
 ) : JwtTokenProvider {
 
   private val logger = LoggerFactory.getLogger(JwtTokenProviderImpl::class.java)
 
   private val key: SecretKey
     get() = Keys.hmacShaKeyFor(jwtSecretProvider.jwtSecret)
 
-  override fun generateToken(userId: Long): JwtToken {
+  /**
+   * This generates a JWT token.
+   * isSuper = true makes this key able to do sensitive operations (is JWT)
+   * Super JWT can be returned only when 2FA is passed or is not set
+   */
+  override fun generateToken(userId: Long, isSuper: Boolean): JwtToken {
+    val expirationTime = if (isSuper)
+      currentDateProvider.date.time + authenticationProperties.jwtSuperExpiration
+    else null
+
+    return generateToken(
+      userId,
+      expirationTime
+    )
+  }
+
+  /**
+   * This generates a JWT token.
+   * superExpiration = when this tokens superiority expires
+   * Super JWT can be returned only when 2FA is passed or is not set
+   */
+  override fun generateToken(userId: Long, superExpiration: Long?): JwtToken {
+    val claims = mutableMapOf<String, Any>()
+    if (superExpiration != null) {
+      // super key is needed for sensitive operations
+      claims["superExpiration"] = superExpiration
```

Comment:
it should be named "sex", SuperEXpiration.... This is long and makes the jwt too long

Reply:
And there should be some constant for that...

Reply:
That naming might need some revision...... unless it's deliberate............ :eyes: